js/bitopro.js

```diff
@@ -0,0 +1,814 @@
+                            [ this.parseNumber ('300000000'), this.parseNumber ('0.0013') ],
+                            [ this.parseNumber ('550000000'), this.parseNumber ('0.0012') ],
+                            [ this.parseNumber ('1300000000'), this.parseNumber ('0.0011') ],
+                        ],
+                        'maker': [
+                            [ this.parseNumber ('0'), this.parseNumber ('0.001') ],
+                            [ this.parseNumber ('3000000'), this.parseNumber ('0.00097') ],
+                            [ this.parseNumber ('5000000'), this.parseNumber ('0.0007') ],
+                            [ this.parseNumber ('30000000'), this.parseNumber ('0.0006') ],
+                            [ this.parseNumber ('300000000'), this.parseNumber ('0.0005') ],
+                            [ this.parseNumber ('550000000'), this.parseNumber ('0.0004') ],
+                            [ this.parseNumber ('1300000000'), this.parseNumber ('0.0003') ],
+                        ],
+                    },
+                },
+            },
+            'exceptions': {
+                'exact': {},
+                'broad': {},
+            },
+            'commonCurrencies': {
+            },
+        });
+    }
+
+    async fetchCurrencies (params = {}) {
+        const response = await this.publicGetProvisioningCurrencies (params);
+        const currencies = this.safeValue (response, 'data', []);
+        //
+        //     {
+        //         "data":[
+        //             {
+        //                 "currency":"eth",
+        //                 "withdrawFee":"0.007",
+        //                 "minWithdraw":"0.001",
+        //                 "maxWithdraw":"1000",
+        //                 "maxDailyWithdraw":"2000",
+        //                 "withdraw":true,
+        //                 "deposit":true,
+        //                 "depositConfirmation":"12"
+        //             }
+        //         ]
+        //     }
+        //
+        const result = {};
+        for (let i = 0; i < currencies.length; i++) {
+            const currency = currencies[i];
+            const id = this.safeString (currency, 'currency');
+            const code = this.safeString (currency, 'currency');
+            const deposit = this.safeValue (currency, 'deposit');
+            const withdraw = this.safeValue (currency, 'withdraw');
+            const fee = this.safeNumber (currency, 'withdrawFee');
+            const withdrawMin = this.safeNumber (currency, 'minWithdraw');
+            const withdrawMax = this.safeNumber (currency, 'maxWithdraw');
+            const limits = {
+                'withdraw': {
+                    'min': withdrawMin,
+                    'max': withdrawMax,
+                },
+                'amount': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+            };
+            result[code] = {
```

Comment:
I think you should not use this exchange-provided code directly to avoid storing exchange specifics in a unified structure. You can do something like this:

```
 const currencyId = this.safeString (currency, 'currency');
 const code = this.safeCurrencyCode (currencyId);
```